### ISSUE TITLE
Ability to mark persisted dead letter messages as replayable

### ISSUE THREAD

**Issue by jeremydmiller** [JasperFx/wolverine#135]
Here's my thoughts:

* Need a way to easily move failed messages from deal letter storage back to incoming when something in the system is fixed
* *An* easy thing to do is to add a 0/1 column to the dead letter table for "replayable" (or better names we think of later)
* A new timed operation within the DurabilityAgent to move replayable, dead letter messages back into incoming with status=Incoming and owner = any node

**Comment by jeremydmiller** [comment]
So a few notes, because I think we have a volunteer for this one:

* I'm thinking you just add a bit flag to the dead letter tables that means "this should be replayed". 
* See the `DeadLettersTable` in both the Sql Server & Postgresql projects. 
* As for the background process to find and move over those recoverable dead letter envelopes, see the `DurabilityAgent`. Specifically look at the `IDurabilityAction` and the simpler instances. You'll just be executing a SQL statement to copy the marked records from the dead letter queue table back to the incoming table inside a new `IDurabilityAction` type. The same DurabilityAgent will replay the incoming envelopes when it gets to them
* `owner_id` should be 0 to denote "anybody can pick this up"
* `attempts` should be set back to 0
* `status` should be "Incoming"
* I'd just schedule the new action to happen from the scheduled job timer

**Comment by kebin** [comment]
If I understand correctly, the workflow from user application's perspective is something like:

- Retry exhausts. Failed message ends up in the `DeadLettersTable`.
- User will toggle `replayable` column to `true` when the bug is fixed manually in the db now, perhaps some command tool/UI later.
- `DurabilityAgent` will move the message from `DeadLettersTable` to `Inbox`.

**Comment by jeremydmiller** [comment]
Yep, just that. Pretty crude, hopefully easy enough to use for a relatively sophisticated shop.

**Comment by kebin** [comment]
Added a `MoveReplayableErrorMessagesToIncoming : IDurabilityAction` action and was able to move `replayable` messages from `DeadLettersTable` with sql. 

I did notice that `DeadLettersTable` does not have `received_at` column set for error messages. Incoming messages require this column, so I set it to `TransportConstants.RepliesUri` during the action( Is this the correct way though?). With this in place, the action was picked up by `DurabilityAgent` and retried.

I will try writing tests and submit a pull request.
